src/network.rs

```diff
@@ -1246,6 +1248,39 @@ impl<'a> Network<'a> {
     }
   }
 
+  async fn get_album_for_track(&mut self, track_id: String) {
+    match self.spotify.track(&track_id).await {
+      Ok(track) => {
+        // It is unclear when the id can ever be None, but perhaps a track can be album-less. If
+        // so, there isn't much to do here anyways, since we're looking for the parent album.
+        let album_id = match track.album.id {
+          Some(id) => id,
+          None => return,
+        };
+
+        if let Ok(album) = self.spotify.album(&album_id).await {
+          // The way we map to the UI is zero-indexed, but Spotify is 1-indexed.
+          let zero_indexed_track_number = track.track_number - 1;
+          let selected_album = SelectedFullAlbum {
+            album,
+            // Overflow should be essentially impossible here, so we prefer the cleaner 'as'.
+            selected_index: zero_indexed_track_number as usize,
+          };
+
+          let mut app = self.app.lock().await;
+
+          app.selected_album_full = Some(selected_album.clone());
+          app.saved_album_tracks_index = selected_album.selected_index;
```

Comment:
This was something that was necessary to get the auto-focus to move to the selected track.

That said, I think there may be some improvement we can make regarding this global variable. From what I can tell, it seems like for `Simplified` `AlbumTableContext`'s, we intuitively use `app.selected_album_simplified.selected_index`, whereas for the `Full` case, we instead use this special `app.saved_album_tracks_index`. This was quite confusing to me and led me to wondering why my change without this L1273 was not working. I think the fact that I had this moment of confusion is probably a sign of a code smell.

I gave a shot myself at removing this and relying on `app.selected_album_full`, and it actually seems to have worked! However, since it was a bit larger a fix/improvement, I did not want to pollute the diff here.

In closing, @Rigellute, please let me know if you think the work of getting rid of this global index state is worth it, and I'll take a crack at it.

Reply:
Yeah, tracking indexes of the UI is pretty messy at the moment. I've long wanted to go back an look for a better solution. I think the `tui` crate might have newer ways of dealing with this, but I need to check it out.